### DISCUSSION TITLE
On-demand Revalidation in all dynamic paths from specific route

### DISCUSSION THREAD

**Discussion by joao-pedrozo** [vercel/next.js#38292] [Ideas]
### Describe the feature you'd like to request

Today is possible to revalidate ONE route with res.revalidate()
```
// pages/api/revalidate.js

export default async function handler(req, res) {
    await res.revalidate('/path-to-revalidate')
    return res.json({ revalidated: true })
}
```

But it would be interesting if we could revalidate all paths from a dynamic route, like in the exemple below

```
// pages/api/revalidate.js

export default async function handler(req, res) {
    await res.revalidate('/posts/*')
    return res.json({ revalidated: true })
}
```

### Describe the solution you'd like

Accept a regex or a pattern in res.revalidate() so we could revalidate multiple paths at once.

### Describe alternatives you've considered

Accept regex or a pattern in params to trigger the bulk revalidation.

**Comment by 11koukou** [comment]
Yes that would be useful indeed. However it seems demanding in terms of performance, not sure though

**Comment by rashidul0405** [comment]
On-demand ISR doesn't quite make sense without regex. For example, if we have a header menu change (coming from CMS), it should invalidate all pages. Do we have any support for that? @leerob

**Comment by EryouHao** [comment]
Hi, does the revalidate method now support regex?

like this, `res.revalidate('/posts/*')`

**Comment by bhaveyx** [comment]
Hi, any update regarding this? This would be very helpful @leerob

**Comment by leerob** [comment]
Good news, this is now possible through "cache tags" in the Next.js App Router: https://nextjs.org/docs/app/api-reference/functions/revalidateTag

Rather than invalidating multiple paths, you would tag multiple related fetches, and then revalidate them all at once.

**Reply by jmattiace** [reply]
@leerob does `revalidateTag()` work with cached pages and [on-demand revalidation](https://nextjs.org/docs/pages/building-your-application/data-fetching/incremental-static-regeneration#on-demand-revalidation)? How can tags be assigned to pages?

**Reply by JayaKrishnaNamburu** [reply]
But, for `revalidateTag` we need to upgrade to next@13 and that breaks for `revalidatePaths` which has a open issue here
https://github.com/vercel/next.js/issues/49861

**Reply by leerob** [reply]
@jmattiace `revalidateTag` will require you to incrementally adopt the App Router. It does not work with the Pages Router.

@JayaKrishnaNamburu I'm not sure that issue is related here, it doesn't appear to have a reproducible example.

**Reply by MohsinDev369** [reply]
I am a little lost on this so plz help me out, The data is coming from sanity cms which is embedded in my Nextjs 13 app. 
`
fetchData.fetch(
    groq`*[_type == "product"]{
      _id,
      name,
      "slug": slug.current,
      "src": image.asset->url,
      quantity,
      "category":*[_id==^.category._ref][0].name,
      price,
      details,
      care
    }`,
`
How can I use revalidateTag in this? Any help will be appreciated. The idea is to call revalidateTag after I added a product in sanity cms so the new product can be seen on the rest of the e-commerce website.

**Reply by lucaspieran** [reply]
im using pages router an existing project it can be possible here?

**Reply by norio-club** [reply]
https://nextjs.org/docs/app/api-reference/functions/revalidateTag

**Reply by Adil1109** [reply]
We can use fetch function inside server actions, if i use next: and give a tag like ['subject'] then in another server action i create a subject and call revalitedTag('subject'). it does revalidates data on my machine but not in production in vercel!!!!!!!!!!!!
